lib/shopify_api/rest.ex

```diff
@@ -11,25 +11,81 @@ defmodule ShopifyAPI.REST do
   alias ShopifyAPI.JSONSerializer
   alias ShopifyAPI.REST.Request
 
-  @doc false
-  def get(%AuthToken{} = auth, path, params \\ []) do
-    Request.perform(auth, :get, path, "", params)
+  @doc """
+  Underlying utility retrieval function. The options passed affect both the
+  return value and, ultimately, the number of requests made to Shopify.
+
+  ## Options
+
+    `:pagination` - Can be `:none`, `:stream`, or `:auto`. Defaults to :auto
+    `:auto` will block until all the pages have been retrieved and concatenated together.
+    `:none` will only return the first page. You won't have access to the headers to manually
+      paginate.
+    `:stream` will return a `Stream`, prepopulated with the first page.
+  """
+  @spec get(AuthToken.t(), path :: String.t(), keyword(), keyword()) ::
+          {:ok, %{required(String.t()) => [map()]}} | Enumerable.t()
+  def get(%AuthToken{} = auth, path, params \\ [], options \\ []) do
+    case pagination(options) do
+      :none ->
+        with {:ok, response} <- Request.perform(auth, :get, path, "", params) do
+          {:ok, fetch_body(response)}
```

Comment:
Not something to resolve right now, but making a note of it for later: this changeset has very conveniently put us _really_ close to being able to return the direct `%HTTPoison.Response{}` struct -- which could be helpful for implementing calls to more the more error-prone of Shopify's APIs where more manual control is required.